modules/network/src/integrationTest/java/org/apache/ignite/network/scalecube/ITScaleCubeNetworkMessagingTest.java

```diff
@@ -88,13 +92,37 @@ public void messageWasSentToAllMembersSuccessfully() throws Exception {
             alice.messagingService().weakSend(member, testMessage);
         }
 
-        boolean done = latch.await(3, TimeUnit.SECONDS);
-        assertTrue(done);
+        boolean messagesReceived = messageReceivedLatch.await(3, TimeUnit.SECONDS);
+        assertTrue(messagesReceived);
 
         //Then: All members successfully received message.
         assertThat(getLastMessage(alice), is(testMessage));
         assertThat(getLastMessage(bob), is(testMessage));
         assertThat(getLastMessage(carol), is(testMessage));
+
+        CountDownLatch aliceShutdownLatch = new CountDownLatch(1);
+
+        bob.topologyService().addEventHandler(new TopologyEventHandler() {
+            /** {@inheritDoc} */
+            @Override public void onAppeared(ClusterNode member) {
+                // No-op.
+            }
+
+            /** {@inheritDoc} */
+            @Override public void onDisappeared(ClusterNode member) {
+                if (aliceName.equals(member.name()))
+                    aliceShutdownLatch.countDown();
+            }
+        });
+
+        alice.shutdown();
+
+        boolean aliceShutdownReceived = aliceShutdownLatch.await(3, TimeUnit.SECONDS);
+        assertTrue(aliceShutdownReceived);
+
+        Collection<ClusterNode> networkMembers = bob.topologyService().allMembers();
```

Comment:
I think this code should be extracted into a separate test

Reply:
I think this test is useless anyway and will be removed once we have a proper networking

Reply:
I would suggest splitting it anyway